test/commands/test_commands.go

```diff
@@ -442,7 +445,7 @@ func (self *TestCommands) RemoveRemote(name gitdomain.Remote) {
 	self.MustRun("git", "remote", "rm", name.String())
 }
 
-// RemoveUnnecessaryFiles trims all files that aren'self necessary in this repo.
+// RemoveUnnecessaryFiles trims all files that aren't necessary in this repo.
```

Comment:
For small cleanups like this, my habit is to commit them to the branch where I notice them. This is because at work, each new PR means we pay for more CI minutes and I have to deal with our flaky CI more often.

So, I don't know if I would use this command very often.

This idea sounds like a niche version of `git town hack --beam` (#3338). Maybe we could add a `--propose` flag to `git town hack` (used like `git town hack --beam --propose`)?

Also FYI, you can use `git switch -` (or `git checkout -`) to switch to the previous branch. I use this often when moving between branches. So for example, I might run:

```
git town hack quick-fix
git commit
git town propose
git switch -
```

Reply:
When I've been in such environments, I was often able to persuade the decision-makers that investing a few thousand or ten thousand dollars annually into better CI infrastructure actually saves money. If developers become just 5% more efficient, the cost savings quickly add up given that developer salaries typically run into hundreds of thousands of dollars per year per person. It's also a good way to improve morale and keep high performers.

The `--propose` flag on `git town hack --beam` sounds ideal for users who commit via another tool, like their IDE. They can extract and propose unrelated commits with a single command. I've created https://github.com/git-town/git-town/issues/4377 to track this idea.

I've also created https://github.com/git-town/git-town/issues/4376 for a command that commits directly into a new branch. Let's continue the discussion there. 

Thanks for the inspiration!